lib/spack/spack/compilers/msvc.py

```diff
@@ -119,16 +203,25 @@ def platform_toolset_ver(self):
         """
         return self.msvc_version[:2].joined.string[:3]
 
-    @property
-    def cl_version(self):
-        """Cl toolset version"""
+    def _compiler_version(self, compiler):
+        """Returns version object for given compiler"""
         return Version(
             re.search(
                 Msvc.version_regex,
-                spack.compiler.get_compiler_version_output(self.cc, version_arg=None),
+                spack.compiler.get_compiler_version_output(compiler, version_arg=None, ignore_errors=True),
```

Comment:
(question) why is `ignore_errors=True` now? I would assume this would have to succeed for us to actually have the version (which we would need to proceed).

Reply:
Because we're not just passing the path to `cl` anymore, which exits with `0` regardless of whether it's given input files, this method is use to detect both `ifx` and `cl` versions (since they both have the exact same heuristics for determining version), and for whatever reason, `ifx` will not exit without failing if you don't give it an input file. 
If the invocation actually fails, we're still catching that with the regex parsing failing to obtain a group. If you'd like I can add an `ignore_errors` argument to this method that we supply conditionally for `ifx` vs `cl`?

Reply:
> If you'd like I can add an ignore_errors argument to this method that we supply conditionally for ifx vs cl?

Not needed, but IMO a comment would be useful. This appears to have a similar intent as the `ignore_version_errors` attribute (but we're reaching into this via another pathway).